src/Tools/dotnet-monitor/CollectionRules/ActionOptionsDependencyAnalyzer.cs

```diff
@@ -0,0 +1,255 @@
+
+                PropertyDependency propertyDependency = null;
+
+                while ((foundIndex = newValue.IndexOf(ActionReferencePrefix, startIndex, StringComparison.OrdinalIgnoreCase)) >= 0)
+                {
+                    int suffixIndex = newValue.IndexOf(SubstitutionSuffix, foundIndex, StringComparison.OrdinalIgnoreCase);
+                    if (suffixIndex == -1)
+                    {
+                        _ruleContext.Logger.InvalidActionReferenceToken(options?.Name ?? actionIndex.ToString(), property.Name);
+                        break;
+                    }
+                    startIndex = suffixIndex;
+
+                    string actionAndResult = newValue[(foundIndex + ActionReferencePrefix.Length)..suffixIndex];
+
+                    if (!GetActionResultReference(actionAndResult, actionIndex, out CollectionRuleActionOptions dependencyOptions, out string actionResultName))
+                    {
+                        continue;
+                    }
+
+                    if (propertyDependency == null)
+                    {
+                        propertyDependency = GetOrCreateDependency(actionIndex, property, originalValue);
+                    }
+
+                    var dependency = new ActionDependency(dependencyOptions, actionResultName);
+                    propertyDependency.ActionDependencies.Add(dependency);
+                }
+            }
+        }
+
+        private PropertyDependency GetOrCreateDependency(int actionIndex, PropertyInfo propertyInfo, string originalValue)
+        {
+            if (!_dependencies.TryGetValue(actionIndex, out Dictionary<string, PropertyDependency> properties))
+            {
+                properties = new Dictionary<string, PropertyDependency>();
+                _dependencies.Add(actionIndex, properties);
+            }
+            if (!properties.TryGetValue(propertyInfo.Name, out PropertyDependency dependency))
+            {
+                dependency = new PropertyDependency(propertyInfo, originalValue);
+                properties.Add(propertyInfo.Name, dependency);
+            }
+            return dependency;
+        }
+
+        private bool GetActionResultReference(string actionReference, int actionIndex,
+            out CollectionRuleActionOptions action, out string actionResultName)
+        {
+            action = null;
+            actionResultName = null;
+
+            string[] parts = actionReference.Split(Separator);
+            if (parts.Length != 2)
+            {
+                _ruleContext.Logger.InvalidActionReference(actionReference);
+                return false;
+            }
+
+            string name = parts[0];
+            //We only check previous actions for our dependencies.
+            CollectionRuleActionOptions dependencyOptions = _ruleContext.Options.Actions.Take(actionIndex)
+                .FirstOrDefault(a => a.Name.Equals(name, StringComparison.OrdinalIgnoreCase));
```

Comment:
Action name is not required; use static string.Equals method instead of instance method.